### ISSUE TITLE
docker.Image: expose `contextDigest` to be able to trigger other pulumi ressource updates on build update

### ISSUE THREAD

**Issue by jlabedo** [pulumi/pulumi-docker#633]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details
When building Image with `skipPush: true` there is no way to detect a new build has occurred, as `repoDigest` is not defined. It would be helpful to expose `contextDigest` (or something else) to trigger updates on dependent resources. For example:
```js
const builderImage = new docker.Image("build", {
  build: {
    context: ".",
    dockerfile: "Dockerfile",
    platform: "linux/amd64",
  },
  imageName: "local/image:latest",
  skipPush: true,
});

const commandFromDocker = new command.local.Command(
  "export",
  {
    create: pulumi.interpolate`docker run ${builderImage.imageName}`,
  },
  { dependsOn: builderImage }
);
```
There is no way to trigger an update on the resource `commandFromDocker` when there is a new image build `builderImage`. With this feature, the code could be changed with:
```js
/* ... */
const commandFromDocker = new command.local.Command(
  "export",
  {
    create: pulumi.interpolate`docker run ${builderImage.imageName}`,
    triggers: [builderImage.contextDigest]
  },
  { dependsOn: builderImage }
);
```

### Affected area/feature

docker provider

**Comment by guineveresaenger** [comment]
Hi @jlabedo - thank you for opening this enhancement request.

Would  it work for you in the meantime to use a timestamp or commit SHA to tag `imageName` to trigger name updates on every image build, so that the run command would automatically update as a function of the name being different?

**Comment by jlabedo** [comment]
Hello @guineveresaenger , sure it would work, but how would you do that ?
If i change the image name in 'dockerBuilder', it will update the ressource even if there is no change. Or i am missing something ?

**Comment by guineveresaenger** [comment]
Maybe I'm missing something in your use case! 

I'm suggesting you change the image name based on timestamp or git commit in `builderImage`. This will always trigger an update, including if there is no build context change.

However, the provider heavily relies on existing cache and this should run very quickly in cases where it's a matter of re-tagging the existing image with a new timestamp.

**Comment by jlabedo** [comment]
Well, yes, I can trigger my command with a timestamp:
```ts
const commandFromDocker = new command.local.Command(
  "export",
  {
    create: pulumi.interpolate`TIME=${Date.now().toString()} docker run ${builderImage.imageName}`,
  },
  { dependsOn: builderImage }
);
```

But it will trigger even if there is no need for an update. Moreover, if this is run on CI, the `builderImage` will not be present, and the command will try to run docker on a missing image (because I did not pushed the image).

**Comment by guineveresaenger** [comment]
I'm curious how you are running this container in CI without building the image.

I was reading your situation as a run of `pulumi up` from a local environment, and that you would like to find a way to trigger the `docker run` to happen when your `latest` image was re-built, but not otherwise.

The workaround I'm suggesting is changing the image tag to a timestamp, because that would result in an update to your `docker run` command. While it would necessitate updating docker.Image every time, it will be a quick build due to local caching.

```
const builderImage = new docker.Image("build", {
  build: {
    context: ".",
    dockerfile: "Dockerfile",
    platform: "linux/amd64",
  },
  imageName: pulumi.interpolate`local/image:TIME=${Date.now().toString()}`,
  skipPush: true,
});
```

That

**Comment by lephuongbg** [comment]
It was very convenient with @pulumi/docker version 3 where the image name also contains the hash.

Now with version 4 we have to jump through hoops to do that.

**Comment by guineveresaenger** [comment]
Closed as completed via #739.
